packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx

```diff
@@ -112,6 +116,17 @@ export class ModalBase extends BaseComponent<IModalProps, IDialogState> implemen
     }
   }
 
+  public componentDidMount() {
+    // Not all modals show just by updating their props. Some only render when they are mounted and pass in
+    // isOpen as true. We need to add the keyUp handler in componentDidMount if we are in that case.
+    if (this.state.isOpen && this.state.isVisible && !this.state.hasRegisteredKeyUp) {
+      this._events.on(window, 'keyup', this._onKeyUp, true /* useCapture */);
```

Comment:
We end up calling this in 2 places with similar conditional logic beforehand, but not identical. I can imagine a case in the future where someone will edit one of these locations but not the other. Would you be open to moving the body of `componentDidMount` to a private method that's called unconditionally from both `componentDidMount` and `UNSAFE_componentWillReceiveProps` ? That way future changes will be sure to set `hasRegisteredKeyUp` correctly and any other boolean logic will be encapsulated.

Reply:
yeah sure, I just saw this now so let me do one more iteration

Reply:
updated now, thanks for the suggestion!